### ISSUE TITLE
[CI] SimpleNioTransportTests.testHandlerIsInvokedOnConnectionClose failure

### ISSUE THREAD

**Issue by bleskes** [elastic/elasticsearch#26855]
This seems like a legitimate failure where SimpleNioTransport does not behave as expected. It reproduces with: 

```
gradle :test:framework:test -Dtests.seed=21E4DC920A4E8E0D -Dtests.class=org.elasticsearch.transport.nio.SimpleNioTransportTests -Dtests.method="testHandlerIsInvokedOnConnectionClose"
-Dtests.security.manager=true -Dtests.locale=de-AT -Dtests.timezone=Canada/Newfoundland
```

Giving:

```
   > Throwable #1: java.nio.channels.ClosedChannelException
   >    at __randomizedtesting.SeedInfo.seed([21E4DC920A4E8E0D:5DD080BB00E195C1]:0)
   >    at sun.nio.ch.SocketChannelImpl.setOption(SocketChannelImpl.java:174)
   >    at sun.nio.ch.SocketChannelImpl.setOption(SocketChannelImpl.java:43)
   >    at org.elasticsearch.transport.nio.NioTransport.closeChannels(NioTransport.java:110)
   >    at org.elasticsearch.transport.TcpTransport$NodeChannels.close(TcpTransport.java:444)
   >    at org.elasticsearch.test.transport.MockTransportService$FilteredConnection.close(MockTransportService.java:778)
   >    at org.elasticsearch.test.transport.MockTransportService$5.close(MockTransportService.java:803)
   >    at org.elasticsearch.transport.AbstractSimpleTransportTestCase.testHandlerIsInvokedOnConnectionClose(AbstractSimpleTransportTestCase.java:2181)
   >    at java.lang.Thread.run(Thread.java:745)
```

@tbrooks8 do you mind taking a look?

**Comment by bleskes** [comment]
CI link https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-windows-compatibility/199/console

**Comment by tbrooks8** [comment]
Thanks @bleskes . I was able to reproduce, so I'll investigate.

**Comment by tbrooks8** [comment]
Fixed by d80ad7f0975c7d1cee8f52528589f193364d33cf.
